### PR TITLE
Report settings closed from callback

### PR DESCRIPTION
**Pull request by Asa-Dawson**
Send a stats event with settings~closed when the settings screen is closed

sendStatsEvent in Genie appears to report the current screen itself, so shouldn't need any extra calls beforehand?

### CODE REVIEW

**Comment by Leonie-** [comment]
Settings closed click stat and the settings page stat should be in CAGE. The only stat needed for settings in Genie is the page stat on settings closed, when we return to whatever page we were on.

File: src/core/settings.js [hunk 1]
```diff
@@ -15,6 +15,7 @@ export const create = () => {
         name: "settings-closed",
         callback: () => {
             setAccessibleLayer(true);
+            gmi.setStatsScreen("settings");
```
Comment (Leonie-) [inline src/core/settings.js]: Think we are missing a test here.

Comment (Asa-Dawson) [reply src/core/settings.js]: Will add! I'm summing up all the tests I need to write

File: test/core/settings.test.js [hunk 2]
```diff
@@ -101,6 +105,21 @@ describe("Settings", () => {
             expect(publishConfig.channel).toBe(expectedSignal.channel);
             expect(publishConfig.name).toBe(expectedSignal.name);
         });
+
+        test("sets the stats screen to settings when settings is opened", () => {
```
Comment (Leonie-) [inline test/core/settings.test.js]: This piece of logic needs removing, as it should be in CAGE.

File: test/core/settings.test.js [hunk 3]
```diff
@@ -101,6 +105,21 @@ describe("Settings", () => {
...
+            settings.show(mockGame);
+            expect(mockGmi.setStatsScreen).toHaveBeenCalledWith("settings");
+        });
+
+        test("sets the stats screen back when settings is closed", () => {
```
Comment (Leonie-) [inline test/core/settings.test.js]: This is looking good 👍

**Comment by Leonie-** [comment]
I think this is ready for go now 🌴

**Pull request merged**